Source/DafnyStandardLibraries/src/DafnyStdLibs/DisableNonLinearArithmetic/Strings.dfy

```diff
@@ -0,0 +1,279 @@
+/**
+The Strings module enables converting between numbers such as nat and int, and String
+ */
+module DafnyStdLibs.Strings {
+  import opened Wrappers
+  import opened Arithmetic.Power
+  import opened Arithmetic.Logarithm
+  import Arithmetic
+
+  /**
+  Refine this module to use it.
+  The refinee must define:
+    - the type of character used, Char
+    - a conversion from nat to Char and vice versa
+   */
+  abstract module ParametricConversion refines Arithmetic.LittleEndianNat {
+    import opened Wrappers
+
+    type Char(==)
+    type String = seq<Char>
+
+    type CharSeq = chars: seq<Char> | |chars| > 1 witness *
```

Comment:
Should `CharSeq` also have the restriction that its elements are unique?

Reply:
The deeper point is that there's no specification that actually requires that! I think that's fine as having these utilities available in Dafny is important even if not all properties are proven, but I did cut https://github.com/dafny-lang/dafny/issues/4795 to track that for later.